### PR TITLE
Use Horde instead Swarm

### PR DESCRIPTION
**Pull request by sleipnir**
Hi! First great job with Swarm.

But I would like to suggest an alternative to using Swarm using Horde.
Horde provides more features and a more uniform interface (using Elixir's own standards as Registry) in addition to using a global Supervisor for the cluster.
Larger arguments for Horde can be found at `application_supervisor.ex`

### CODE REVIEW

File: lib/cluster_user/accounts/user_dynamic_supervisor.ex [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+  end
+
+  def start_link(opts) do
+    HordeDynamicSupervisor.start_link(__MODULE__, opts, name: __MODULE__)
   end
 
   def register(user) do
     child_spec = %{
       id: ClusterUser.Accounts.User.Actor,
+      name: user,
```
Comment (eliasdarruda) [inline lib/cluster_user/accounts/user_dynamic_supervisor.ex:18]: What is this `name: user` for?

**Comment by eliasdarruda** [comment]
This is so nice, it changes almost nothing on the original application with a dynamic supervisor and vanilla registry, making the business logic of the application agnostic to swarm which was for me a huge downside

About deploying and having new pods with new code, in sync through libcluster with old pods with old code, would it impact the communication between them? lets say a Actor restarted through an old code supervisor, some service could call some functions that is not implemented in the old code and it would crash...

This crashed call would only be restarted by the old supervisor or it would pass the actor to a new supervisor afterwards?

**Comment by sleipnir** [comment]
> This is so nice, it changes almost nothing on the original application with a dynamic supervisor and vanilla registry, making the business logic of the application agnostic to swarm which was for me a huge downside
> 

Yes I agree.

> About deploying and having new pods with new code, in sync through libcluster with old pods with old code, would it impact the communication between them? lets say a Actor restarted through an old code supervisor, some service could call some functions that is not implemented in the old code and it would crash...
> 
> This crashed call would only be restarted by the old supervisor or it would pass the actor to a new supervisor afterwards?

We would have to simulate this behavior to be sure, but if the process is global it would be started with the new code, as long as the new process has the old methods it would guarantee some backward compatibility. Your concern is with the rolling updates correct? Another approach would be to change the name of the process identifier, this would avoid ambiguous calls.

File: lib/cluster_user/accounts/user_dynamic_supervisor.ex (hunk 1 above)
Comment (sleipnir) [reply lib/cluster_user/accounts/user_dynamic_supervisor.ex:18]: It sure wasn't supposed to be there :D

**Review by wesleimp — APPROVED** [review]
💜

**Pull request merged**